### DISCUSSION TITLE
Rhino Json schema validator library

### DISCUSSION THREAD

**Discussion by dapiced** [mozilla/rhino#1466] [Q&A] [answered]
Does Rhino have a Json schema validator library? if not is there a way to install it? Especially with VMware aria orchestrator integration
if rhino does not accept adding module, should i use nodejs ou python in aria orchestrator  then ?

**Comment by StSchnell** [comment, marked as answer]
Hello Dominic (@dapiced),

in the VMware Aria Orchestrator you have also the possibility to use different runtime environments. Beside JavaScript is PowerShell available. PowerShell contains a cmdlet call [Test-Json](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/test-json?view=powershell-7.4). With this cmdlet it is very easy to realize a JSON schema validator. Here the PowerShell action:

```powershell
function Handler($context, $inputs) {
    $result = Test-Json -Json $inputs.in_jsonContent -Schema $inputs.in_jsonSchema;
    $output=@{status = 'done'; result = $result};
    return $output
}
```

![image](https://github.com/mozilla/rhino/assets/130429818/d1307184-289a-4039-8e75-00394a276563)

Now you can call this action from your JavaScript environment. Here I use the Microsoft example from the Test-Json description. Here the JavaScript action:

```javascript
var jsonContent = '{"name": "Ashley", "age": 25}';

var jsonSchema = '{ \
  "definitions": {}, \
  "$schema": "[http://json-schema.org/draft-07/schema#"](http://json-schema.org/draft-07/schema#%22), \
  "$id": "[http://example.com/root.json"](http://example.com/root.json%22), \
  "type": "object", \
  "title": "The Root Schema", \
  "required": [ \
    "name", \
    "age" \
  ], \
  "properties": { \
    "name": { \
      "$id": "#/properties/name", \
      "type": "string", \
      "title": "The Name Schema", \
      "default": "", \
      "examples": [ \
        "Ashley" \
      ], \
      "pattern": "^(.*)$" \
    }, \
    "age": { \
      "$id": "#/properties/age", \
      "type": "integer", \
      "title": "The Age Schema", \
      "default": 0, \
      "examples": [ \
        25 \
      ] \
    } \
  } \
}';

var result = System.getModule("de.stschnell").jsonSchemaValidator(
  jsonContent,
  jsonSchema
);

return result.result;
```

![image](https://github.com/mozilla/rhino/assets/130429818/53e221ee-2eaf-4f35-9a4c-3e05d1494d05)

With this approach it is not necessary to install an additional library. You simply combine the existing possibilities of VMware Aria Automation. Because as far as I know a JSON schema validator is not available.

Best regards
Stefan

**Reply by dapiced** [reply]
thank you very much !
